database/mssql/scripts/dbscripts/PSP_PIMS_S31_00/Alter Up/01_PSP_PIMS_S30_00_Alter_Up.sql

```diff
@@ -0,0 +1,1087 @@
+	[APP_LAST_UPDATE_USERID]        	nvarchar(30) NOT NULL CONSTRAINT [NOTE_APP_LAST_UPDATE_USERID_DEF]  DEFAULT (user_name()),
+	[APP_LAST_UPDATE_USER_GUID]     	uniqueidentifier NULL,
+	[APP_LAST_UPDATE_USER_DIRECTORY]	nvarchar(30) NOT NULL CONSTRAINT [NOTE_APP_LAST_UPDATE_USER_DIRECTORY_DEF]  DEFAULT (user_name()),
+	[DB_CREATE_TIMESTAMP]           	datetime NOT NULL CONSTRAINT [NOTE_DB_CREATE_TIMESTAMP_DEF]  DEFAULT (getutcdate()),
+	[DB_CREATE_USERID]              	nvarchar(30) NOT NULL CONSTRAINT [NOTE_DB_CREATE_USERID_DEF]  DEFAULT (user_name()),
+	[DB_LAST_UPDATE_TIMESTAMP]      	datetime NOT NULL CONSTRAINT [NOTE_DB_LAST_UPDATE_TIMESTAMP_DEF]  DEFAULT (getutcdate()),
+	[DB_LAST_UPDATE_USERID]         	nvarchar(30) NOT NULL CONSTRAINT [NOTE_DB_LAST_UPDATE_USERID_DEF]  DEFAULT (user_name()),
+	CONSTRAINT [NOTE_PK] PRIMARY KEY CLUSTERED([NOTE_ID])
+ ON [PRIMARY])
+GO
+EXEC sp_addextendedproperty 
+	@name = N'MS_Description', @value = N'Contents of the note.' , 
+	@level0type = N'Schema', @level0name = N'dbo', 
+	@level1type = N'Table', @level1name = N'PIMS_NOTE', 
+	@level2type = N'Column', @level2name = N'NOTE_TXT'
+GO
+EXEC sp_addextendedproperty 
+	@name = N'MS_Description', @value = N'Table to contain all applicable notes for the PIMS PSP system.' , 
+	@level0type = N'Schema', @level0name = N'dbo', 
+	@level1type = N'Table', @level1name = N'PIMS_NOTE'
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Create sequence dbo.PIMS_NOTE_H_ID_SEQ
+PRINT N'Create sequence dbo.PIMS_NOTE_H_ID_SEQ'
+GO
+CREATE SEQUENCE [dbo].[PIMS_NOTE_H_ID_SEQ]
+	AS bigint
+	START WITH 1
+	INCREMENT BY 1
+	MINVALUE 1
+	MAXVALUE 2147483647
+	NO CYCLE
+	CACHE 50
+GO
+IF @@ERROR <> 0 SET NOEXEC ON
+GO
+
+-- Create table dbo.PIMS_NOTE_HIST
+PRINT N'Create table dbo.PIMS_NOTE_HIST'
+GO
+CREATE TABLE [dbo].[PIMS_NOTE_HIST]  ( 
+	[_NOTE_HIST_ID]                 	bigint NOT NULL DEFAULT (NEXT VALUE FOR [PIMS_NOTE_H_ID_SEQ]),
+	[EFFECTIVE_DATE_HIST]           	datetime NOT NULL DEFAULT (getutcdate()),
+	[END_DATE_HIST]                 	datetime NULL,
+	[NOTE_ID]                       	bigint NOT NULL,
+	[CONCURRENCY_CONTROL_NUMBER]    	bigint NOT NULL,
+	[APP_CREATE_TIMESTAMP]          	datetime NOT NULL,
+	[APP_CREATE_USERID]             	nvarchar(30) NOT NULL,
+	[APP_CREATE_USER_GUID]          	uniqueidentifier NULL,
+	[APP_CREATE_USER_DIRECTORY]     	nvarchar(30) NOT NULL,
+	[APP_LAST_UPDATE_TIMESTAMP]     	datetime NOT NULL,
+	[APP_LAST_UPDATE_USERID]        	nvarchar(30) NOT NULL,
+	[APP_LAST_UPDATE_USER_GUID]     	uniqueidentifier NULL,
+	[APP_LAST_UPDATE_USER_DIRECTORY]	nvarchar(30) NOT NULL,
+	[DB_CREATE_TIMESTAMP]           	datetime NOT NULL,
+	[DB_CREATE_USERID]              	nvarchar(30) NOT NULL,
+	[DB_LAST_UPDATE_TIMESTAMP]      	datetime NOT NULL,
+	[DB_LAST_UPDATE_USERID]         	nvarchar(30) NOT NULL,
+	CONSTRAINT [PIMS_NOTE_H_PK] PRIMARY KEY CLUSTERED([_NOTE_HIST_ID])
+ ON [PRIMARY])
```

Comment:
another question @dfilteau - why are we not capturing the actual note text (NOTE_TXT) in the history table. Seems to me all we are capturing is that a note with this ID got changed on this date but not the actual change? please clarify

Reply:
That's an issue with the MoTI scripts.  They don't allow NVARCHAR(max) to be supported in history tables.  I'll investigate and set to an allowable value - maybe 4000.  Probably because NVARCHAR(max) is a CLOB - must impact performance and/or storage constraints.

Reply:
Cancelling the PR until the NOTE_TXT issue is fixed